scripts/prerelease.js

```diff
@@ -0,0 +1,77 @@
+/* This script will turn all packages in the repo into a prerelease with the existing version */
+const npmFetch = require('npm-registry-fetch');
+const fs = require('fs');
+const path = require('path');
+const semver = require('semver');
+
+const rootFolder = path.join(__dirname, '../');
+const rushConfig = JSON.parse(fs.readFileSync(path.join(rootFolder, 'rush.json')).toString());
+
+const newVersions = {};
+const preleaseTag = 'beta';
+
+function semverSort(a, b) {
+  if (semver.gt(a, b)) {
+    return 1;
+  }
+
+  if (semver.lt(a, b)) {
+    return -1;
+  }
+
+  return 0;
+}
+
+(async () => {
+  await rushConfig.projects.reduce(async (previousPromise, project) => {
+    await previousPromise;
+
+    const projectFolder = path.join(rootFolder, project.projectFolder);
+    const packageJson = JSON.parse(fs.readFileSync(path.join(projectFolder, 'package.json')));
+
+    let npmInfo = {};
+
+    try {
+      const response = await npmFetch(`/${packageJson.name}`);
+      npmInfo = await response.json();
+    } catch (e) {
+      console.log(`cannot fetch ${packageJson.name} from public npm registry`);
+    }
+
+    if (npmInfo.versions) {
+      const checkedInVersion = packageJson.version;
+      const prereleaseVersions = Object.keys(npmInfo.versions).filter(version => version.startsWith(checkedInVersion + '-' + preleaseTag));
+      const latestVersion =
+        prereleaseVersions.length > 0
+          ? prereleaseVersions.sort(semverSort)[prereleaseVersions.length - 1]
```

Comment:
The version bump approach (or strategy) is VERY VERY specific. I think you should have a big Jsdoc segment at the top of this file fully documenting what the script does, with some examples, and documenting any arguments if any (it looks like it takes in no arguments).

Reply:
Good call. It was meant to be something we do a bit more "internally" as related to part of the release, but this can be a bit more generalized to be useful actually.